database/src/main/kotlin/no/nav/su/se/bakover/database/RevurderingPostgresRepo.kt

```diff
@@ -0,0 +1,178 @@
+    private val dataSource: DataSource,
+    private val behandlingRepo: BehandlingRepo
+) : RevurderingRepo {
+    override fun hent(id: UUID): Revurdering? =
+        dataSource.withSession { session ->
+            "select * from revurdering where id = :id"
+                .hent(mapOf("id" to id), session) { row ->
+                    row.toRevurdering()
+                }
+        }
+
+    override fun lagre(revurdering: Revurdering) {
+        when (revurdering) {
+            is OpprettetRevurdering -> lagre(revurdering)
+            is BeregnetRevurdering -> lagre(revurdering)
+            is SimulertRevurdering -> lagre(revurdering)
+            is RevurderingTilAttestering -> lagre(revurdering)
+        }
+    }
+
+    fun Row.toRevurdering(): Revurdering {
+        val id = uuid("id")
+        val periode = string("periode").let { objectMapper.readValue<Periode>(it) }
+        val opprettet = tidspunkt("opprettet")
+        val tilRevurdering = behandlingRepo.hentBehandling(uuid("behandlingId"))!!
+        val beregning = stringOrNull("beregning")?.let { objectMapper.readValue<PersistertBeregning>(it) }
+        val simulering = stringOrNull("simulering")?.let { objectMapper.readValue<Simulering>(it) }
+        val saksbehandler = string("saksbehandler")
+        val oppgaveId = stringOrNull("oppgaveId")
+        return when {
+            oppgaveId != null && simulering != null && beregning != null -> RevurderingTilAttestering(
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering,
+                beregning = beregning,
+                simulering = simulering,
+                saksbehandler = Saksbehandler(saksbehandler),
+                oppgaveId = OppgaveId(oppgaveId)
+            )
+            simulering != null && beregning != null -> SimulertRevurdering(
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering,
+                beregning = beregning,
+                simulering = simulering,
+                saksbehandler = Saksbehandler(saksbehandler)
+            )
+            beregning != null -> BeregnetRevurdering(
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering, beregning = beregning, saksbehandler = Saksbehandler(saksbehandler)
+            )
+            else -> OpprettetRevurdering(
+                id = id,
+                periode = periode,
+                opprettet = opprettet,
+                tilRevurdering = tilRevurdering,
+                saksbehandler = Saksbehandler(saksbehandler)
+            )
+        }
+    }
+
+    private fun lagre(revurdering: OpprettetRevurdering) =
+        dataSource.withSession { session ->
+            (
+                "insert into revurdering" +
```

Comment:
Hvorfor ikke multiline? https://realkotlin.com/tutorials/2018-06-26-multiline-string-literals-in-kotlin/